frontend/src/services/validate-csv.service.ts

```diff
@@ -62,8 +62,18 @@ export async function validateCsv({
             preview = hydrateTemplate(template, row, removeEmptyLines)
           }
         } catch (e) {
+          // Exit early if it is a templating error. This can be caused by either the template or first row having invalid values.
```

Comment:
If we want to validate every single row for an invalid image source at this stage, we will have to run `templateClient.template` for every row. This is will be an additional duplicate call for each row before the processing done later on in `protectAndUploadCsv`.